### PR TITLE
python3Packages.compiledb: move to pkgs/by-name

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kirillrdy** [comment]
@Sigmanificient you need to remove entry from `python-packages.nix`

**Review by multun — APPROVED** [review]
LGTM, I'd rather be removed from maintainers as I don't use nix anymore

**Comment by Sigmanificient** [comment]
@multun done, you may also want to do a pr on your side to fully remove yourself from `maintainers-list.nix` and some other python packages left

*PS: hello form France :D*

**Review by multun — APPROVED** [review]

**Review by multun — APPROVED** [review]

File: pkgs/by-name/co/compiledb/package.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  lib,
+  fetchFromGitHub,
+  coreutils,
+  python3Packages,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "compiledb";
+  version = "0.10.1";
+
```
Comment (FliegendeWurst) [inline pkgs/by-name/co/compiledb/package.nix:11]:
```suggestion
  pyproject = true;
```

File: pkgs/by-name/co/compiledb/package.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  '';
+
+  nativeCheckInputs = [ python3Packages.pytestCheckHook ];
+
+  doCheck = true;
+
+  meta = {
+    description = "Tool for generating Clang's JSON Compilation Database files";
+    mainProgram = "compiledb";
+    license = lib.licenses.gpl3;
```
Comment (FliegendeWurst) [inline pkgs/by-name/co/compiledb/package.nix:40]:
```suggestion
    license = lib.licenses.gpl3Plus;
```
Per https://github.com/nickdiego/compiledb/blob/b615d526019c7b809882090ea7eaff85423f2b13/compiledb/parser.py#L8-L11 and LICENSE combined

**Review by multun — APPROVED** [review]

**Pull request merged**